jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -1601,6 +1601,190 @@ ecma_builtin_typedarray_prototype_find_index (ecma_value_t this_arg, /**< this a
   return ecma_builtin_typedarray_prototype_find_helper (this_arg, predicate, predicate_this_arg, false);
 } /* ecma_builtin_typedarray_prototype_find_index */
 
+/**
+ * The %TypedArray%.prototype object's 'copyWithin' routine helper
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static uint32_t
+ecma_builtin_typedarray_prototype_copy_within_helper(uint32_t length, /**< length of the typedarray */
+                                                     ecma_number_t index, /**< copyWithin target or start or end property */
+                                                     bool is_end_index ) /**< true - normalize the end index */
+{
+  uint32_t ret_value;
+
+  if (!ecma_number_is_nan (index))
+  {
+    if (ecma_number_is_zero (index))
+    {
+      ret_value = 0;
+    }
+    else if (ecma_number_is_infinity (index))
+    {
+      ret_value = ecma_number_is_negative (index) ? 0 : length;
+    }
+    else
+    {
+      if (ecma_number_is_negative (index))
+      {
+        ecma_number_t index_neg = -index;
+        int32_t norm_index = (int32_t) ((int32_t)length - (int32_t) ecma_number_to_int32 (index_neg));
+        ret_value = norm_index > 0 ? (uint32_t) norm_index : 0;
+      }
+      else
+      {
+        ret_value = ecma_number_to_uint32 (index) > length ? length : ecma_number_to_uint32 (index);
```

Comment:
In this case the `index` is converted twice (the to_uint32 call). If possible only one call should be done.

Reply:
Is this a min value calculation? If so the `JERRY_MIN` macro could help to make this more explicit. (There is also `JERRY_MAX` for the code part above.)